scripts/validate_docstrings.py

```diff
@@ -494,6 +494,15 @@ def method_source(self):
         except TypeError:
             return ''
 
+    @property
+    def clean_method_source(self):
```

Comment:
Still think adding a property here overcomplicates things. I know you are trying to guard against `return` showing up in a comment which is fine, but is that not possible to do just in the pattern itself?

Reply:
It is possible, I was actually doing that a few commits back.

The `clean_method_source` property that I wrote currently makes 4 operations on the method source. Each one of those operations allowed me to make a simplification in the regex pattern, as can be seen individually in each of these commits: 2051ea6, ff17a79 and bafc25d.

To lose this property would mean having to go back to the 3 line verbose regex pattern I was using before, that was way too complicated. Be it as it may, these complexities must go somewhere, otherwise the algorithm will fail on some cases (already covered by the tests). 

One possibility would be to merge the `clean_method_source` property with the already existing `method_source` property. So far, it's only being used twice (at least in this .py file), to search for the returns (at the point I'm trying to update) and to search for yields. I'm not sure if that would be semantically correct though.

Reply:
-1 on this approach (using regex), I guess we should be able to use the AST to know whether something can be returned by the function.

Reply:
I removed this property.